From bf3adf0525811d83bd8a418ae78dbfc56dccfd31 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Tue, 13 Dec 2022 11:49:20 +0700
Subject: [PATCH] fdpm: fix issue failed to get "fdp1-[12]" clock

Clock "fdp1-[12]" are on different nodes with "fdp1-0", so using the
same platform_device to get these clocks will cause issue.
Correct the platform_device can fix this issue.

Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drv/fdpm_sub.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drv/fdpm_sub.c b/drv/fdpm_sub.c
index 0b99fa3..cb19c1f 100755
--- a/drv/fdpm_sub.c
+++ b/drv/fdpm_sub.c
@@ -190,7 +190,7 @@ int fdpm_init(struct fdpm_privdata *priv)
 	}
 
 #if defined(M2CONFIG) || defined(H2CONFIG)
-	pdrv->fdpm_clk[1] = clk_get(&pdev->dev, "fdp1-1");
+	pdrv->fdpm_clk[1] = clk_get(&pdev_c2->dev, "fdp1-1");
 	if (IS_ERR(pdrv->fdpm_clk[1])) {
 		APRINT("failed to get FDP1 clock\n");
 		ercd = -EFAULT;
@@ -205,7 +205,7 @@ int fdpm_init(struct fdpm_privdata *priv)
 #endif
 
 #if defined(H2CONFIG)
-	pdrv->fdpm_clk[2] = clk_get(&pdev->dev, "fdp1-2");
+	pdrv->fdpm_clk[2] = clk_get(&pdev_c3->dev, "fdp1-2");
 	if (IS_ERR(pdrv->fdpm_clk[2])) {
 		APRINT("failed to get FDP2 clock\n");
 		ercd = -EFAULT;
-- 
2.25.1

